src/requests/consumption.js

```diff
@@ -0,0 +1,744 @@
+			components: [],
+			resourceDetails,
+			isAPrivateProgram: false,
+			isDeleted: false,
+			requestForPIIConsent: templateData?.requestForPIIConsent ? true : false,
+			rootOrganisations: [
+				templateData?.rootOrganisations ? templateData?.rootOrganisations : templateData?.organization?.id,
+			],
+			createdFor: [templateData?.createdFor ? templateData?.createdFor : templateData?.organization?.id],
+			deleted: false,
+			status: common.STATUS_ACTIVE.toLowerCase(),
+			owner: templateData?.created_by,
+			createdBy: templateData?.created_by,
+			updatedBy: templateData?.created_by,
+			externalId: generateExternalId(templateData?.title),
+			name: templateData?.title.trim(),
+			description: templateData?.description ? templateData?.description : '',
+			updatedAt: new Date(),
+			createdAt: new Date(),
+			__v: 0,
+			endDate: new Date(templateData?.end_date),
+			startDate: new Date(templateData?.start_date),
+		}
+
+		return { success: true, template }
+	} catch (error) {
+		console.error('Error in formatTemplate:', error.message)
+		return { success: false, error: error.message }
+	}
+}
+
+/**
+ * Publish the Program
+ * @name publishProjectTemplates
+ * @param {Object} programData - Program template data
+ * @returns {Object} - Response of Program creation
+ */
+const publishProgram = function (programData) {
+	return new Promise(async (resolve, reject) => {
+		const result = { success: false, templateId: null, error: null }
+		try {
+			// Format the template
+			let formattedTemplate = formatProgramTemplate(programData)
+			if (!formattedTemplate.success) {
+				throw new Error('FAILED_TO_FORMAT_TEMPLATE')
+			}
+
+			let template = formattedTemplate.template
+			const resourceDetails = template.resourceDetails
+			const programScope = template.scope
+			delete template.resourceDetails
+
+			// Insert the template into the database
+			const programsCollection = mongoDb.collection(COLLECTIONS.PROGRAMS)
+			const result = await programsCollection.insertOne(template)
+
+			// Validate the result of the template creation
+			if (!result || !result.insertedId) {
+				throw new Error('Failed to insert the template into the database.')
+			}
+
+			const programId = result.insertedId
+
+			const duplicateResource = await duplicateResources(resourceDetails)
+
+			const solutions = await createSolutions(duplicateResource, {
+				_id: programId,
+				scope: programScope,
+				externalId: template.externalId,
+				name: template.name,
+				description: template.description ? template.description : '',
+				end_date: template.endDate,
+				start_date: template.startDate,
+			})
+
+			const solutionIds = solutions.map((solution) => solution._id)
+
+			// Update Template with tasks and sequence
+			await programsCollection.updateOne(
+				{ _id: programId },
+				{
+					$set: {
+						components: solutionIds,
+					},
+				}
+			)
+
+			await rolloutService.publishCallback(programData.id, programId.toString())
```

Comment:
@adithyadinesh0412  where we are updating the published_id of the solution in SCP?